### PR TITLE
Preserve user's cause names and ordering in results

### LINKED ISSUE #29: Preserve user's cause names and ordering in results
## Context
Results should reflect the cause names and ordering exactly as they appear in the user's uploaded VA data, rather than using internally standardized names or a default ordering.

## What to do
- Display cause names in results (charts, tables) exactly as they appear in the uploaded VA data
- Maintain the same cause ordering the user provided in their input
- This applies to both the CSMF chart and the misclassification matrix

## Why
Different VA algorithms may use different naming conventions (e.g., "Preterm" vs "Prematurity", "Birth asphyxia" vs "Intrapartum"). Users expect to see their own terminology reflected in the output.

## Source
Dashboard features document (2026-03-02), "Results" section

### PR DESCRIPTION
**Pull request by cliu238**
## Summary
- Backend now builds a mapping from broad cause categories back to the user's original cause names (`cause_display_names`) and preserves first-appearance ordering (`cause_order`) in the result object
- Frontend uses these for display in CSMF charts, CSMF tables, and misclassification matrices, falling back to default names for sample/demo data
- Extracted `formatCauseDisplay()` and `orderCauses()` into a shared utility (`causeDisplay.js`) replacing duplicated `formatCause()` functions

Fixes #29

## Test plan
- [x] 10 new frontend unit tests for `formatCauseDisplay` and `orderCauses` (custom names, fallback, ordering, edge cases)
- [x] 11 new R backend tests for `build_cause_display_map` and `build_cause_order` (InterVA→Prematurity, EAVA→Preterm mappings, ordering)
- [x] All 76 frontend tests pass (up from 66)
- [x] All 104 R tests pass (up from 93)
- [x] Lint clean, build succeeds, integration check passes
- [ ] E2E test: submit EAVA upload, verify results show "Preterm" not "Prematurity" (requires running backend)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Custom cause names from uploaded data are preserved and used for labels in tables, charts, and matrices
  * Original cause ordering from your input is maintained across results

* **Bug Fixes / Improvements**
  * UI consistently falls back to sensible, human-friendly cause names when custom names aren’t provided

* **Tests**
  * Added unit and integration tests covering cause display mappings and ordering

### CODE REVIEW

File: backend/jobs/utils.R [hunk 1]
```diff
@@ -252,6 +252,41 @@ build_broad_matrix <- function(df, age_group) {
...
+      freq <- table(original_causes)
+      result[[broad_cause]] <- names(freq)[which.max(freq)]
+    }
+  }
+  result
+}
+
+# Build cause ordering based on first appearance in user's data.
+# Returns broad cause names ordered by when they first appear in the CSV.
+build_cause_order <- function(df, broad_matrix) {
```
[the remark this answers is not in the document]
Comment (cliu238) [reply backend/jobs/utils.R]: Fixed in da7b550 — removed the unused `df` parameter from `build_cause_order()` and updated all 3 call sites.
